www/tutorial/app/xterm-terminal/xterm-terminal.component.ts

```diff
@@ -0,0 +1,171 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+declare var Terminal: any;
+
+@Component({
+  selector: 'xterm-terminal',
+  templateUrl: 'app/xterm-terminal/xterm-terminal.component.html',
+  styleUrls: ['app/xterm-terminal/xterm-terminal.component.css']
+})
+export class XtermTerminalComponent implements OnInit {
+  @Input() responsesArray: any;
+  @Output() stepNumber: EventEmitter<number> = new EventEmitter<number>();
+  step: number = 0;
+  shellprompt: string;
+  buffer: string = '';
+  terminalContainer: any;
+  term: any;
+  optionElements: any;
+  cols: string;
+  rows: string;
+  previousCommands: any = [];
+
+ ngOnInit() {
+    this.terminalContainer = document.getElementById('terminal-container'),
+    this.optionElements = {
+      cursorBlink: true
+    },
+    this.cols = '70',
+    this.rows = '70';
+    this.createTerminal();
+  }
+
+  createTerminal() {
+    while (this.terminalContainer.children.length) {
+      this.terminalContainer.removeChild(this.terminalContainer.children[0]);
+    }
+    this.term = new Terminal({
+      cursorBlink: true
+    });
+
+    this.term.open(this.terminalContainer);
+    this.term.fit();
+
+    var initialGeometry = this.term.proposeGeometry(),
+      cols = initialGeometry.cols,
+      rows = initialGeometry.rows;
+
+    this.cols = cols;
+    this.rows = rows;
+    this.runFakeTerminal()
+  }
+
+  runFakeTerminal() {
+    if (this.term._initialized) {
+      return;
+    }
+    this.term._initialized = true;
+    this.setPrompt();
+  }
+
+  setPrompt() {
```

Comment:
we should move out the special handling for inspec to app.component.ts. This leaves us with a reusable xterm component